### DISCUSSION TITLE
DocsContextProps args deprecated.

### DISCUSSION THREAD

**Discussion by borodovisin** [storybookjs/storybook#17155] [General] [answered]
I want to know if in further versions we won't have available the args in the context object.
I mean using this code:

```

Story.parameters = {
    docs: {
       container: ({ context }) => {
            // Something here...
           context.args is deprecated // This will disappear in the future
       }
   }
}
```
On version 6.4.9 `DocsContextProps` has args mark as deprecated.

**Comment by shilman** [comment]
Yes, the plan is to remove them in v7. Per this comment in the `DocsContextProps` type definition:

```
  // These keys are deprecated and will be removed in v7
  /** @deprecated */
  kind?: ComponentTitle;
  /** @deprecated */
  story?: StoryName;
  /** @deprecated */
  args?: Args;
  /** @deprecated */
  globals?: Globals;
  /** @deprecated */
  parameters?: Globals;
```

**Reply by borodovisin** [reply]
There is a way to get those values in version 7. I have a case right now for a library where I'm using my own custom Doc page and I generate my code based on the args. meaning I return my component in the `docs.container` function. In the example above inside the function, I return my React component. 
I need the args there to generate my code and set it on the `@codesandbox/sandpack-react`. 

That means in version 7 I won't be able to get access to `context.args` and I'll lose the ability to generate dynamic code to make the final user life a little easier, coz now they can play with the controls and copy the generated code without coding at all.

**Reply by camdenlee** [reply]
I'm using the context in a similar fashion and would love more insight into the changes and how access to that data will be available in v7.

**Reply by shilman** [reply]
cc @tmeasday

**Reply by tmeasday** [reply]
The docs context has `initialArgs`, which is the args for the story as defined in the CSF file. 

<strike>My understanding is the context does not change when args values are changed e.g. via controls so I am pretty sure the args here will go out of date. This is existing behaviour in previous versions of SB I think.</strike> <- this was wrong I guess!

Did you want the dynamically changing args or was it just the `initialArgs` you needed?

**Reply by borodovisin** [reply]
Hi @tmeasday, answering your question; the dynamic values. I need to capture all the changes on the controls. The context inside the example above has the changed args. Meaning for my case any change on the controls is reflected on my `@codesandbox/sandpack-react` custom component.

In the controls let say I have a field value `initialArgs` equal to 10. When the users play with that field and change the value to 20 I'm able to get that change in the `context.args` and generate my code dynamically based on that interaction. If the `args` will be deprecated how can I get the changed values as a result of some user interaction with the controls?

**Reply by tmeasday** [reply]
Have you tried `docsContext.getStoryContext(docsContext.storyById(docsContext.id))`?

**Reply by borodovisin** [reply]
@tmeasday that means the safe way to get access to the `args` key is using this:

`docsContext.getStoryContext(docsContext.storyById(docsContext.id))?.args`

I guess from here: `StoryContextUpdate` right? Just to confirm 🎯

**Reply by tmeasday** [reply]
I think so! Sorry I am a bit hazy on the details at the moment but my recollection is we wanted to take the "current" story's details off the context as it was confusing when you are probably rendering multiple stories in any given docs page.

**Reply by shilman** [reply]
cc @patricklafrance @frassinier ☝️

**Reply by borodovisin** [reply]
Hi, I following the types on the latest version.
`getStoryContext` return `StoryContextForLoaders<TFramework>`
and `StoryContextForLoaders` type is:
`StoryContextForLoaders<TFramework extends AnyFramework = AnyFramework, TArgs = Args> = StoryContextForEnhancers<TFramework, TArgs> & Required<StoryContextUpdate<TArgs>>`

Inside I found `StoryContextUpdate` type:
```
export declare type StoryContextUpdate<TArgs = Args> = {
    args?: TArgs;
    globals?: Globals;
    [key: string]: any;
}
```

That is why I think I can get the `args` using this line: 
`docsContext.getStoryContext(docsContext.storyById(docsContext.id))?.args`

The `args` key is optional, but I guess is the way to get access to the `args` in the control, right?. 
I just want to make sure I'm not taking the long path with a bunch of unnecessary steps to get access to the current state of the `args`.

**Reply by tmeasday** [reply]
@borodovisin - `args` is optional on `StoryContextUpdate`, but the type is `Required<StoryContextUpdate>` so it not optional on `StoryContextForLoaders`.

I think you are doing it right!

**Comment by borodovisin** [comment, marked as answer]
Hi @tmeasday, thanks for all the help on this question 💯 .

**Reply by tmeasday** [reply]
No worries at all! Sorry about the undocumented changes.
